File: frontend/src/features/users/types/Audit.ts
export interface AuditLog {
  id: string;
  userId: string;
  action: string;
  changes: Record<string, { old: string; new: string }>;
  ipAddress?: string;
  userAgent?: string;
  timestamp: string;
}

export interface AuditLogPage {
  items: AuditLog[];
  total: number;
}

File: user-service/internal/core/domain/users/audit.go
package users

import "time"

type AuditLog struct {
	ID        string                  `json:"id"`
	UserID    ID                      `json:"userId"`
	Action    string                  `json:"action"`
	Changes   map[string]ChangeDetail `json:"changes"`
	IPAddress string                  `json:"ipAddress,omitempty"`
	UserAgent string                  `json:"userAgent,omitempty"`
	Timestamp time.Time               `json:"timestamp"`
}

type ChangeDetail struct {
	Old string `json:"old"`
	New string `json:"new"`
}
